### ISSUE TITLE
Setting remote root-dir{work} no longer works with csh

### ISSUE THREAD

**Issue by MartinDix** [metomi/rose#2103]
Our site still has a few users with csh as their login shell.

Our server site configuration sets
```
root-dir{share}=raijin*=/short/$PROJECT/$USER
root-dir{share/cycle}=raijin*=/short/$PROJECT/$USER
root-dir{work}=raijin*=/short/$PROJECT/$USER
```

With rose 2017.01.0 the command that's run was
```
 ssh -oBatchMode=yes -oConnectTimeout=10 raijin.nci.org.au bash --login -c \'ROSE_VERSION=2017.01.0\ /projects/access/bin/rose\ suite-run\ -v\ -v\ --name=au-aa398\ --install-only\ --run=run\ --remote=uuid=8f2d273a-8420-4486-ba73-3907d85de521,root-dir{share}=/short/$PROJECT/$USER,root-dir{share/cycle}=/short/$PROJECT/$USER,root-dir{work}=/short/$PROJECT/$USER\'
```
This meant that ```{share}``` etc was interpreted by bash and everything worked ok.

WIth 2015.05.0 (and remote-no-login-shell=false) the command that's run is
```
ssh -oBatchMode=yes -oConnectTimeout=10 -n raijin.nci.org.au env\ ROSE_VERSION=2017.05.0\ CYLC_VERSION=7.4.0\ bash\ -l\ -c\ \'\"$0\"\ \"$@\"\'\ /projects/access/bin/rose\ suite-run\ -vv\ -n\ au-aa398\ --install-only\ --run=run\ --remote=uuid=df365ae4-cd0f-424a-9c39-a950bdb9cd68,root-dir{share}=/short/$PROJECT/$USER,root-dir{share/cycle}=/short/$PROJECT/$USER,root-dir{work}=/short/$PROJECT/$USER
```
Here it seems that the arguments are interpreted by the remote login shell before being passed to bash and so the {} get dropped by csh. rose then actually gets arguments like root-dirshare and so all the work and share directories end up in the home filesytem.

I think the best approach is to use ```re.escape``` on ```root-dir{share}``` etc. This seems to work with all shells and would mean we could set remote-no-login-shell=true.

Aa far as I can tell, nothing in the rose test-battery tests whether these directories are correctly created remotely. Would it be worth me trying to write a test for this?

**Comment by matthewrmshin** [comment]
`pipes.quote` is probably better than `re.escape` for the purpose.

There are actually tests under `t/rose-suite-clean/` to test that the directories are correctly created remotely (so that the tests can prove that they get removed correctly as well). However, I don't think we have ever tested these with a remote login shell as `csh` or `tcsh`.

Please feel free to propose fixes and tests.

**Comment by aosprey** [comment]
We are also having an issue with this, although I am using bash. 

We set
```
root-dir=login*.archer.ac.uk=$DATADIR
```

And the command that's reported is:
```
ssh -oBatchMode=yes -n login.archer.ac.uk env\ ROSE_VERSION=2017.10.0\ CYLC_VERSION=7.5.0\ bash\ -l\ -c\ \'\"$0\"\ \"$@\"\'\ rose\ suite-run\ -vv\ -n\ archer-test1\ --debug\ --no-log-archive\ --run=run\ --remote=uuid=ca142de7-dd0d-4d97-96e5-a0ab9b6b9345,now-str=20171018T111639Z,root-dir=$DATADIR
```

However, the `$DATADIR` variable is being evaluated on the suite host, and as our suite host is totally independent of the remote HPC, this variable is unset.  (I can see this with verbose flags in the ssh command).

Matt's solution of `pipes.quote` works perfectly though.

**Comment by aosprey** [comment]
This is a big issue for us, as it doesn't work under any shell as far as I can see. So we would be keen to get this fix in. 

Martin, does the `pipes.quote` fix for you too? I am happy to put in a branch but I don't want to step on toes if you've been looking at this.

**Comment by sadielbartholomew** [comment]
I am looking to close down Issues that may no longer be relevant.

@MartinDix, please can you let us know whether Annette's fix as above worked/works for your site's users using ``csh`` (or some similar shell)? Or indeed whether they still use that as their login shell?

If not, do you still intend to add some tests for a (``t``)``csh`` remote login shell? Otherwise, I believe those shell choices are so uncommon now, across our user-base at least, that we can close this Issue. Thanks.

**Comment by matthewrmshin** [comment]
Assume fixed by #2194.
